### PR TITLE
macos and collector pre-generation testing

### PR DESCRIPTION
**Pull request by michaelsembwever**

### CODE REVIEW

File: ds-collector-tests/cluster-dse-k8s.make [hunk 1]
```diff
@@ -8,16 +8,21 @@ test:
 	echo "" >> /tmp/datastax/test-collector-k8s-dse.conf
 	echo "cqlshUsername=$$(kubectl -n cass-operator get secret cluster2-superuser -o yaml | grep " username" | awk -F" " '{print $$2}' | base64 -d && echo "")" >> /tmp/datastax/test-collector-k8s-dse.conf
 	echo "cqlshPassword=$$(kubectl -n cass-operator get secret cluster2-superuser -o yaml | grep " password" | awk -F" " '{print $$2}' | base64 -d && echo "")" >> /tmp/datastax/test-collector-k8s-dse.conf
-	../ds-collector/ds-collector -T -f /tmp/datastax/test-collector-k8s-dse.conf -n cluster2-dc1-default-sts-0
-	../ds-collector/ds-collector -X -f /tmp/datastax/test-collector-k8s-dse.conf -n cluster2-dc1-default-sts-0
+	./collector/ds-collector -T -f /tmp/datastax/test-collector-k8s-dse.conf -n cluster2-dc1-default-sts-0
```
Comment (AndrewHogg) [inline ds-collector-tests/cluster-dse-k8s.make:11]: I'm missing something on this pathing change.

File: ds-collector-tests/cluster-dse-k8s.make [hunk 2]
```diff
@@ -8,16 +8,21 @@ test:
...
 	if ! ls /tmp/datastax/ | grep -q ".tar.gz" ; then echo "Failed to generate artefacts in the K8s cluster "; ls -l /tmp/datastax/ ; exit 1 ; fi
 
 
 setup:
-	rm -f ../ds-collector/collect-info
-	cd ../ds-collector ; docker run --rm -v $$PWD:/volume -w /volume -t clux/muslrust rustc --target x86_64-unknown-linux-musl rust-commands/*.rs ; cd -
-	test -f ../ds-collector/collect-info
 	mkdir -p /tmp/datastax && rm -fr /tmp/datastax/*
+	# make diagnostics bundle
+	cd ../ ; ISSUE="TEST-$$(git rev-parse --abbrev-ref HEAD)--$$(git rev-parse --short HEAD)" make
```
Comment (AndrewHogg) [inline ds-collector-tests/cluster-dse-k8s.make:20]: Not sure if its the branch name, but this causes a failure later in the sed : sed: 1: "s/\#issueId=.*/issueId= ...": bad flag in substitute command: 'h'

**Review by AndrewHogg — CHANGES_REQUESTED** [review]

File: ds-collector-tests/cluster-dse-k8s.make (hunk 2 above)
Comment (michaelsembwever) [reply ds-collector-tests/cluster-dse-k8s.make:20]: odd. not in the github actions. i guess you are seeing this locally…? on macos? 
( i noticed in github actions that `$(git rev-parse --abbrev-ref HEAD)` is always `HEAD`. so maybe when the branch name actually comes in, the `/` slash crashes the sed…

File: ds-collector-tests/cluster-dse-k8s.make [hunk 3]
```diff
@@ -8,16 +8,21 @@ test:
 	echo "" >> /tmp/datastax/test-collector-k8s-dse.conf
 	echo "cqlshUsername=$$(kubectl -n cass-operator get secret cluster2-superuser -o yaml | grep " username" | awk -F" " '{print $$2}' | base64 -d && echo "")" >> /tmp/datastax/test-collector-k8s-dse.conf
 	echo "cqlshPassword=$$(kubectl -n cass-operator get secret cluster2-superuser -o yaml | grep " password" | awk -F" " '{print $$2}' | base64 -d && echo "")" >> /tmp/datastax/test-collector-k8s-dse.conf
-	../ds-collector/ds-collector -T -f /tmp/datastax/test-collector-k8s-dse.conf -n cluster2-dc1-default-sts-0
-	../ds-collector/ds-collector -X -f /tmp/datastax/test-collector-k8s-dse.conf -n cluster2-dc1-default-sts-0
+	./collector/ds-collector -T -f /tmp/datastax/test-collector-k8s-dse.conf -n cluster2-dc1-default-sts-0
+	./collector/ds-collector -T -p -f /tmp/datastax/test-collector-k8s-dse.conf -n cluster2-dc1-default-sts-0
```
[the remark this answers is not in the document]
Comment (michaelsembwever) [reply ds-collector-tests/cluster-dse-k8s.make:12]: just the ping check yes.

[the remark this answers is not in the document]
Comment (michaelsembwever) [reply ds-collector-tests/cluster-dse-k8s.make:12]: (just testing both with and without ping)

File: ds-collector-tests/cluster-dse-k8s.make (hunk 2 above)
Comment (michaelsembwever) [reply ds-collector-tests/cluster-dse-k8s.make:20]: should be fixed with https://github.com/datastax/diagnostic-collection/pull/90/commits/7a32ad8e4d6a2bd67511ab3c83eae2f8c4312af6

(this is a good bug to catch)

Comment (AndrewHogg) [reply ds-collector-tests/cluster-dse-k8s.make:20]: Yes, I checked out the branch and was running the tests locally with a make -f cluster-dse-k8s.make

Comment (michaelsembwever) [reply ds-collector-tests/cluster-dse-k8s.make:20]: try now…

Comment (AndrewHogg) [reply ds-collector-tests/cluster-dse-k8s.make:20]: that now works - not sure if there are other characters we need to be wary of on this, but more of an edge case at that point.

File: ds-collector-tests/cluster-one-node-vanilla-ssh-docker.make [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+test:
+	# ds-collector over SSH
+	docker exec -t ds-collector-tests_bastion_1 /collector/ds-collector -T -f /ds-collector-tests/test-collector-ssh.conf -n ds-collector-tests_cassandra-00_1
+	docker exec -t ds-collector-tests_bastion_1 /collector/ds-collector -T -p -f /ds-collector-tests/test-collector-ssh.conf -n ds-collector-tests_cassandra-00_1
+	docker exec -t ds-collector-tests_bastion_1 /collector/ds-collector -X -f /ds-collector-tests/test-collector-ssh.conf -n ds-collector-tests_cassandra-00_1
+	# test archives exist
+	if ! ( docker exec ds-collector-tests_bastion_1 ls /tmp/datastax/ ) | grep -q ".tar.gz" ; then echo "Failed to generate artefacts in the SSH cluster" ; ( docker exec ds-collector-tests_bastion_1 ls /tmp/datastax/ ) ; exit 1 ; fi
+	# ds-collector over docker
+	./collector/ds-collector -T -f test-collector-docker.conf -n ds-collector-tests_cassandra-00_1
+	./collector/ds-collector -T -p -f test-collector-docker.conf -n ds-collector-tests_cassandra-00_1
```
Comment (AndrewHogg) [inline ds-collector-tests/cluster-one-node-vanilla-ssh-docker.make:14]: as discussed, the ping can fail and at this stage we do not know if it fails because of a script issue, or fails because the ping itself fails. the return code is non-zero but it does not exit.

**Review by AndrewHogg — APPROVED** [review]

**Comment by AndrewHogg** [comment]
Approved
